### PR TITLE
fix early deletion of wasm state schema

### PR DESCRIPTION
**Pull request by kyessenov**
Signed-off-by: Kuat Yessenov <kuat@google.com>

For an explanation of how to fill out the fields, please see the relevant section
in [PULL_REQUESTS.md](https://github.com/envoyproxy/envoy/blob/master/PULL_REQUESTS.md)

Commit Message:
Additional Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Deprecated:]

### CODE REVIEW

**Comment by kyessenov** [comment]
The return result changed, but it's not checked by metadata exchange plugin at the moment. Behaviourally it's the same as long as double registration happens to be identical.

File: source/extensions/common/wasm/context.cc [hunk 1]
```diff
@@ -950,8 +950,12 @@ WasmResult Context::setProperty(absl::string_view path, absl::string_view value)
 
 WasmResult Context::declareProperty(absl::string_view path,
                                     std::unique_ptr<const WasmStatePrototype> state_prototype) {
-  state_prototypes_[path] = std::move(state_prototype);
-  return WasmResult::Ok;
+  // Do not delete existing schema since it can be referenced by state objects.
+  if (state_prototypes_.find(path) == state_prototypes_.end()) {
+    state_prototypes_[path] = std::move(state_prototype);
+    return WasmResult::Ok;
+  }
```
Comment (mandarjog) [inline source/extensions/common/wasm/context.cc:957]: It should be bad argument only if the new and the old registration is different, right?

Comment (kyessenov) [reply source/extensions/common/wasm/context.cc:957]: That would require equality for the wasm prototypes. In general, it's not clear to me what to do if this method fails in the client. So the client should just log and ignore.

**Comment by mandarjog** [comment]
@PiotrSikora @jplevyak We need this in 1.6...

File: source/extensions/common/wasm/context.cc (hunk 1 above)
Comment (kyessenov) [reply source/extensions/common/wasm/context.cc:957]: I'd like to get rid of protobuf as an argument eventually. The return result is currently ignored.

Comment (jplevyak) [reply source/extensions/common/wasm/context.cc:957]: SGTM.

**Pull request merged**